app/services/facility_stats_service.rb

```diff
@@ -0,0 +1,71 @@
+class FacilityStatsService
+  attr_reader :facilities_data, :stats_by_size
+
+  def initialize(accessible_facilities:, retain_facilities:, ending_period:, rate_numerator:)
+    @facilities = accessible_facilities
+    @retain_facilities = retain_facilities
+    @ending_period = ending_period
+    @rate_numerator = rate_numerator
+    @rate_name = "#{rate_numerator}_rate"
+    @periods = ending_period.downto(5)
+    @facilities_data = {}
+    @stats_by_size = {}
+  end
+
+  def call
+    facilities.each do |facility|
+      facility_data = Reports::RegionService.new(region: facility, period: ending_period).call
```

Comment:
With the above proposal, this call would go away and would become input to the service object.